tools/firmware-utils/src/mkzyxelzldfw.c

```diff
@@ -0,0 +1,835 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Copyright (C) 2020 Vincent Wiemann <vw@derowe.com>
+ *
+ * This program is derived from ZyXEL's GPL U-Boot code.
+ * Copyright (C) 2011-2011 ZyXEL Communications, Corp.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation.
+ *
+ */
+
+#define _POSIX_SOURCE
+#define _POSIX_C_SOURCE 199309L /* getopt */
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <time.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <string.h>
+#include <unistd.h>
+#include <netinet/in.h>
+#include <linux/limits.h>
+#include "md5.h"
+
+#define ZYXEL_MAGIC	0xdeadbeaf
+#define MAX_MODELS	5
+#define CHECKSUM_SIZE	sizeof(uint32_t)
+#define MAX_FILES	32
+#define MAX_FILENAME	64
+
+#define DATE_SIZE	32
+#define REV_SIZE	32
+
+#define error(fmt, ...)                                                        \
+	do {                                                                   \
+		printf("Error: " fmt, ##__VA_ARGS__);                          \
+		exit(1);                                                       \
```

Comment:
Indentation is a bit odd here, but probably nobody cares ...

Reply:
That's clang kernel coding style. I like it. If OpenWrt has other preferences, there should be a clang file.

Reply:
And why does the first line have more trailing spaces than the others?

Reply:
Ah... I think this is because of the hash sign at the beginning. Seems to be a missing case in the clang file.

Reply:
Okay, I'll just forget about it ... :-)

Reply:
@adschm This is only a display issue in Github I think...

Reply:
When I replace the tabs with 8 spaces, it is correct... Very strange...